tutorials/katacoda/3-eventing-intro-channels/step4.md

```diff
@@ -0,0 +1,101 @@
+or functions. We then need to pass the events from the producer through a series of consumers. This time rather than creating several channels and subscriptions, we
+will create a `Sequence` instead. A Sequence lets us define an in-order list of functions that will be invoked. Sequence creates `Channel`s and `Subscription`s under the hood.
+
+Let's see this in action now. We will create the following topology where the event is passed from Producer to Consumer 1, who transforms the event, and then passes it
+along to Consumer 2, which displays the event.
+
+![seq](./assets/sequence.png)
+
+First, let's create the final consumer:
+
+```
+cat <<EOF | kubectl create -f -
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display-chain
+spec:
+  template:
+    spec:
+      containers:
+        - image: gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display
+EOF
+```{{execute}}
+
+Then create the first consumer which will modify the message:
+
+```
+cat <<EOF | kubectl create -f -
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: first
+spec:
+  template:
+    spec:
+      containers:
+        - image: gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/appender
+          env:
+            - name: MESSAGE
+              value: " - Updating this message a little..."
+EOF
+```{{execute}}
+
+Now, let's create the Sequence to execute the service "first" and then pass its result to the service "event-display-chain"
+
+```
+cat <<EOF | kubectl create -f -
+apiVersion: flows.knative.dev/v1beta1
+kind: Sequence
+metadata:
+  name: sequence
+spec:
+  channelTemplate:
+    apiVersion: messaging.knative.dev/v1beta1
+    kind: InMemoryChannel
+  steps:
+    - ref:
+        apiVersion: serving.knative.dev/v1
+        kind: Service
+        name: first
+  reply:
+    ref:
+      kind: Service
+      apiVersion: serving.knative.dev/v1
+      name: event-display-chain
+EOF
+```{{execute}}
+
+Finally, create the producer to create the event:
+
+```
+cat <<EOF | kubectl create -f -
+apiVersion: sources.knative.dev/v1alpha2
+kind: PingSource
+metadata:
+  name: ping-source-sequence
+spec:
+  schedule: "*/1 * * * *"
+  jsonData: '{"message": "Hello Sequence!"}'
+  sink:
+    ref:
+      apiVersion: flows.knative.dev/v1beta1
+      kind: Sequence
+      name: sequence
+EOF
+```{{execute}}
+
+Verify that the sequence was executed and the message was updated by running the below command
+(You will not see an event there for a minute after creating the producer):
+
+```
+# it is likely that is pod is still being created after scaling down to zero
+kubectl wait --for=condition=ready pod -l serving.knative.dev/service=event-display-chain
+# see the logs
+kubectl -n default logs -l serving.knative.dev/service=event-display-chain -c user-container --since=10m --tail=50
+```{{execute}}
```

Comment:
I had to run it twice to see the output of the logs

there was a timeout error also

```
pingsource.sources.knative.dev/ping-source-sequence created
$ # it is likely that is pod is still being created after scaling down to zero
$ kubectl wait --for=condition=ready pod -l serving.knative.dev/service=event-display-chain
pod/event-display-chain-00001-deployment-566755999b-g4xqg condition met
$ # see the logs
$ kubectl -n default logs -l serving.knative.dev/service=event-display-chain -c user-container --since=10m --tail=50
$ # it is likely that is pod is still being created after scaling down to zero
$ kubectl wait --for=condition=ready pod -l serving.knative.dev/service=event-display-chain
error: timed out waiting for the condition on pods/event-display-chain-00001-deployment-566755999b-g4xqg
$ # see the logs
$ kubectl -n default logs -l serving.knative.dev/service=event-display-chain -c user-container --since=10m --tail=50
☁️  cloudevents.Event
Validation: valid
Context Attributes,
  specversion: 1.0
  type: dev.knative.sources.ping
  source: /apis/v1/namespaces/default/pingsources/ping-source-sequence
  id: d65624c9-70dd-4bbd-8ec0-d1bfbfbb39dd
  time: 2021-03-23T01:59:00.103779523Z
  datacontenttype: application/json
Data,
  {
    "id": 0,
    "message": "Hello Sequence! - Updating this message a little..."
  }
```

Reply:
The default timeout was 30 seconds, bumped it to 90.